google/cloud/aiplatform/metadata/execution.py

```diff
@@ -14,95 +14,337 @@
+            resource_id (str):
+                Optional. The <resource_id> portion of the Execution name with
+                the format. This is globally unique in a metadataStore:
+                projects/123/locations/us-central1/metadataStores/<metadata_store_id>/executions/<resource_id>.
+            display_name (str):
+                Optional. The user-defined name of the Execution.
+            schema_version (str):
+                Optional. schema_version specifies the version used by the Execution.
+                If not set, defaults to use the latest version.
+            metadata (Dict):
+                Optional. Contains the metadata information that will be stored in the Execution.
+            description (str):
+                Optional. Describes the purpose of the Execution to be created.
+            metadata_store_id (str):
+                Optional. The <metadata_store_id> portion of the resource name with
+                the format:
+                projects/123/locations/us-central1/metadataStores/<metadata_store_id>/artifacts/<resource_id>
+                If not provided, the MetadataStore's ID will be set to "default".
+            project (str):
+                Optional. Project used to create this Execution. Overrides project set in
+                aiplatform.init.
+            location (str):
+                Optional. Location used to create this Execution. Overrides location set in
+                aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials used to create this Execution. Overrides
+                credentials set in aiplatform.init.
+
+        Returns:
+            Execution: Instantiated representation of the managed Metadata Execution.
+
+        """
+        self = cls._empty_constructor(
+            project=project, location=location, credentials=credentials
+        )
+        super(base.VertexAiResourceNounWithFutureManager, self).__init__()
+
+        resource = Execution._create_resource(
+            client=self.api_client,
+            parent=metadata_store._MetadataStore._format_resource_name(
+                project=self.project,
+                location=self.location,
+                metadata_store=metadata_store_id,
+            ),
+            schema_title=schema_title,
+            resource_id=resource_id,
+            metadata=metadata,
+            description=description,
+            display_name=display_name,
+            schema_version=schema_version,
+            state=state,
+        )
+        self._gca_resource = resource
+
+        return self
+
+    def __enter__(self):
+        if self.state is not gca_execution.Execution.State.RUNNING:
+            self.update(state=gca_execution.Execution.State.RUNNING)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        state = (
+            gca_execution.Execution.State.FAILED
+            if exc_type
```

Comment:
If an exception is raised in the code block being executed within the context manager then `exc_type` is the class of that exception. ie:

```Python
with aiplatform.Execution(...) as exc:
  raise RunTimeError('')
```

Otherwise it is `None`

Reply:
I see. Thanks for the info.